models/spring-ai-openai/src/main/java/org/springframework/ai/openai/OpenAiChatModel.java

```diff
@@ -436,7 +454,7 @@ ChatCompletionRequest createRequest(Prompt prompt, boolean stream) {
 				return List.of(new ChatCompletionMessage(content,
 						ChatCompletionMessage.Role.valueOf(message.getMessageType().name())));
 			}
-			else if (message.getMessageType() == MessageType.ASSISTANT) {
+			else if (MessageType.ASSISTANT.equals(messageType)) {
```

Comment:
Yes, it is an enum. However, it is always safer to use `equals(:Object)` than `==`. Besides, a properly constructed `equals(:Object)` method covers `==` anyway. Let the JIT compiler optimize.

Additionally, while it may not apply in this context, serialization will most certainly invalidate the use of `==`.

I'd argue that `==` should only be used for primitive types, and rarely, if ever Object types.